src/main/kotlin/cacophony/semantic/syntaxtree/SyntaxTree.kt

```diff
@@ -50,12 +58,21 @@ sealed class Type(
                 areEquivalentTypes(argumentsType, other.argumentsType) &&
                 areEquivalentTypes(returnType, other.returnType)
     }
+
+    class Struct(range: Pair<Location, Location>, val fields: Map<String, Type>) : Type(range) {
```

Comment:
maybe this could be `Type.Structural` to match naming convention with `Type.basic` and `Type.Functional`